### PR TITLE
chore/milestone-model

### PR DESCRIPTION
**Pull request by satyavh**
This includes: 
- migrating Milestones to models (not including milestone actions)
- token integration > making it possible to donate in tokens

This needs to be run with the corresponding feathers branch: 
feature/token-integration

And make sure to migrate your local db before running!

### CODE REVIEW

**Comment by vojtechsimetka** [comment]
Deploy preview for *giveth-dapp* processing.

Built with commit 588a915c8ab0f8ef95ff48d71018241e23f795ba

https://app.netlify.com/sites/giveth-dapp/deploys/5bd1a39773f2cf1c937c621b

**Comment by vojtechsimetka** [comment]
Deploy preview for *giveth-dapp* processing.

Built with commit 588a915c8ab0f8ef95ff48d71018241e23f795ba

https://app.netlify.com/sites/giveth-dapp/deploys/5bd1a397c6aed6173edb82ef

File: src/lib/blockchain/getWeb3.js [hunk 1]
```diff
@@ -111,11 +111,45 @@ const setWallet = (rpcUrl, isHomeNetwork = false) =>
     this.setProvider(engine);
   };
 
+/**
+  Method to extend web3 to poll tx receipt mined. Usage:
+  web3.eth.getTransactionReceiptMined = _getTransactionReceiptMined;
+  Gist: https://gist.github.com/xavierlepretre/88682e871f4ad07be4534ae560692ee6
+**/
+const _getTransactionReceiptMined = function(txHash, interval){
```
Comment (perissology) [inline src/lib/blockchain/getWeb3.js]: web3 1.0 already provides this functionality and will only resolve the promise once the tx is mined

File: src/models/MilestoneModel.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
+import React from 'react';
```
Comment (perissology) [inline src/models/MilestoneModel.js]: why the inconsistent naming of models? None of the existing models have the `Model` suffix

File: src/contextProviders/UserProvider.jsx [hunk 3]
```diff
@@ -27,6 +27,7 @@ feathersClient
   .find()
   .then(whitelist => {
     React.whitelist = whitelist;
+    console.log('React.whitelist', whitelist)
```
Comment (perissology) [inline src/contextProviders/UserProvider.jsx]: probably can be removed

File: src/configuration.js [hunk 4]
```diff
@@ -46,7 +46,7 @@ const configurations = {
     tokenAddresses: { 'Ropsten ETH': '0x387871cf72c8CC81E3a945402b0E3A2A6C0Ed38a' },
     etherscan: 'https://rinkeby.etherscan.io/',
     homeEtherscan: 'https://ropsten.etherscan.io/',
-    feathersConnection: 'https://feathers.develop.giveth.io',
+    feathersConnection: 'http://localhost:3030',
```
Comment (perissology) [inline src/configuration.js]: I don't think this should be changed

File: src/components/DonateButton.jsx [hunk 5]
```diff
@@ -149,107 +223,151 @@ class DonateButton extends React.Component {
...
+    const isDonationInToken = selectedToken.symbol !== 'ETH'
+    const tokenAddress = isDonationInToken ? selectedToken.address : 0;
+
+    const _makeDonationTx = async () => {
+      let method;
+      let donationUser;
+      let opts;
+
+      if(isDonationInToken)
+        opts = { from: account, gas: 1000000 };
```
Comment (perissology) [inline src/components/DonateButton.jsx]: this gas is ridiculously high, maybe we can just remove it and let the gas estimator figure it out.

File: package.json [hunk 6]
```diff
@@ -71,7 +71,7 @@
     "prettier": "^1.13.7"
   },
   "scripts": {
-    "precommit": "lint-staged",
+    "prepush": "lint-staged",
```
Comment (perissology) [inline package.json]: Does this work?  The linting only runs on the currently stagged files, but before a push, there will be no stagged files

File: src/components/DonateButton.jsx (hunk 5 above)
Comment (satyavh) [reply src/components/DonateButton.jsx]: Yeah we tried that. It does not work.
It put it on 100.000, it says it uses 84766 but it runs out of gas with the exact number

File: package.json (hunk 6 above)
Comment (satyavh) [reply package.json]: no it works, but it was a great dev experience :-)

File: src/models/MilestoneModel.js (hunk 2 above)
Comment (satyavh) [reply src/models/MilestoneModel.js]: Unfortunately, but we should for clearer declaration (.e.g `Milestone` the component vs `Milestone` the model)
Switched it back for now, don't want to spend time on renaming

File: src/lib/blockchain/getWeb3.js (hunk 1 above)
Comment (satyavh) [reply src/lib/blockchain/getWeb3.js]: I tried some stuff with events but those did not fire. 
But removed and changed to `getTransactionReceipt`

**Pull request merged**